waltz-ng/client/data-types/routes.js

```diff
@@ -0,0 +1,42 @@
+const baseState = {
+    resolve: {
+        dataTypes: [
+            'DataTypesService',
+            (dataTypesService) => dataTypesService.loadDataTypes()
+        ]
+    }
+};
+
+
+const listState = {
+    url: 'data-types',
+    views: {'content@': require('./data-type-list') }
+};
+
+
+const viewByCodeState = {
+    url: 'data-types/code/{code}',
+    views: {'content@': require('./data-type-view') },
+    redirectTo: 'main.data-type.view',
```

Comment:
I don't think the resolver approach would work.  The resolver put the resolved promises into the controllers parameters, whereas we need them in state params to do a redirect.  We also still need the redirect functionality in order to ensure the URL has a id and not the code.

I've been looking at refactoring the redirectTo code to use a function that can be passed in from outside as follows:

 function redirectTo($rootScope, $state, paramResolver) {
    $rootScope.$on('$stateChangeStart', function(evt, to, params) {
        if (to.redirectTo) {
            evt.preventDefault();

            $state.go(to.redirectTo, paramResolver(params));
        }
    });
}

I've managed to come up with a solution, i'll check it in now.  Let me know your thoughts